agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -207,9 +207,7 @@ void slave_thread::on_thread_stop() { stop_slave_thread(); }
 
 bool slave_thread::socket_disconnected(Socket *sd)
 {
-    if (slave_state == STATE_WAIT_FOR_WIFI_CONFIGURATION_UPDATE_COMPLETE ||
-        slave_state == STATE_WAIT_FOR_ANOTHER_WIFI_CONFIGURATION_UPDATE ||
-        slave_state == STATE_WAIT_FOR_UNIFY_WIFI_CREDENTIALS_RESPONSE) {
+    if (configuration_in_progress) {
```

Comment:
I had needed to remove the first two comparisons as part of this commit and replace it with the `configuration_on_process` but since I knew the 3rd comparisons would need to drop as well, I preferred doing it all together.